### PR TITLE
Fixes for topic dropdown

### PR DESCRIPTION
**Pull request by lzach83**
## Link to Issue
Closes: #5660 

## Description of Changes
- Added back the `value` prop to `TopicSelector` 

## "How We Fixed It"

- The `value` prop was missing previously, added it back and works for this use case

## Test Plan
- Ensure `TopicSelector` works everywhere it's used.

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/topic_selector.tsx [hunk 1]
```diff
@@ -44,6 +44,7 @@ export const TopicSelector = ({
       options={allOptions}
       className="TopicSelector"
       onChange={handleOnChange}
+      value={topicToOption(value, false)}
```
Comment (lzach83) [inline packages/commonwealth/client/scripts/views/components/topic_selector.tsx:47]: I assumed uses just `value` would work here but causes multiple down-line type issues. To the best of my knowledge this works with zero regressions, would be interested in if there's another obvious solution... taking the time crunch to launch gating in consideration. 

cc: @rbennettcw @mzparacha

**Review by rbennettcw — APPROVED** [review]
Topic dropdown works for thread create + topic edit

**Review by mzparacha — CHANGES_REQUESTED** [review]
https://github.com/hicommonwealth/commonwealth/assets/51641047/15bd245d-24b2-4a82-8e08-3433055ddad5

mostly works correctly but breaks when we visit the new thread page after visiting the "All" discussions page

**Review by mzparacha — APPROVED** [review]
LGTM -- just need to fix code quality checks

**Pull request merged**